pkg/bundle/read.go

```diff
@@ -247,6 +250,39 @@ func read(ctx context.Context, name, baseDir string, bundleSpecReader io.Reader,
 	return New(def, scans...)
 }
 
+// fixAndApplyHelmSpec checks if the spec's Helm repo, if there is one, ends with a '/' and appends one if not. See chartURL
+// This function also propagates root Helm chart properties to the child targets and performs the repo check.
+// Note: The repo and version properties of child targets are only propagated if either property is overridden. This
+// allows child targets to specify a path to a chart, to specify a url to the chart(e.g. oci://), etc.
+func fixAndApplyHelmSpec(spec *fleet.BundleSpec) {
+	if spec.Helm != nil && spec.Helm.Chart != "" {
+		if spec.Helm.Repo != "" && !strings.HasSuffix(spec.Helm.Repo, "/") {
+			spec.Helm.Repo = spec.Helm.Repo + "/"
+		}
+	}
+
+	for _, target := range spec.Targets {
+		if target.Helm != nil && spec.Helm != nil {
+			if target.Helm.Repo != "" || target.Helm.Version != "" {
+				if target.Helm.Repo == "" {
+					target.Helm.Repo = spec.Helm.Repo
+				}
+				if target.Helm.Version == "" {
+					target.Helm.Version = spec.Helm.Version
+				}
+			}
+
+			if target.Helm.Chart == "" {
+				target.Helm.Chart = spec.Helm.Chart
```

Comment:
This step should be obsolete because it is already handled in `calculate.go`.

Reply:
I think my logic here was by doing the merge here, any place that needed the helm options would be guaranteed to have the finalized helm options. Looking at it now, I think I agree that it's obsolete. I think the above if statement just needs to be added to `calculate.go`

Reply:
No need to add it, it was already there. The `Chart` `targetCustomization` was already supported by fleet, but `Version` and `Repo` not, which you have added in this pr.